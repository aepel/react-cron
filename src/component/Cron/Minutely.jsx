import React, { useState } from 'react'
import { FormControlLabel, FormGroup, Radio, Stack, TextField } from '@mui/material'

function Minutely({ cronExpression, onChange }) {
  const [minutes, setMinutes] = useState(10)
  const onMinuteChange = e => {
    if ((e.target.value > 0 && e.target.value < 60) || e.target.value === '') {
      setMinutes(parseInt(e.target.value, 10))
      const val = [...cronExpression]
      val[0] = `*/${minutes}`
      if (e.target.value === '') {
        val[0] = '*/10'
      } else {
        val[0] = `*/${e.target.value}`
      }

      onChange(val)
    }
  }

  return (
    <Stack direction="column" spacing={2} alignItems="flex-start" justifyContent="flex-start">
      <FormGroup row>
        <FormControlLabel
          control={
            <Radio
              color="primary"
              value="1"
              onChange={() => {
                onChange([cronExpression[0], '*', '*', '*', '*'])
              }}
              checked={cronExpression[4] === '*'}
            />
          }
          label="Every x Minutes"
        />
        <FormControlLabel
          control={
            <Radio
              color="primary"
              value="4"
              onChange={() => {
                onChange([cronExpression[0], '*', '*', '*', '1-5'])
              }}
              checked={cronExpression[4] === '1-5'}
            />
          }
          label="Every X minutes only Weekdays"
        />
      </FormGroup>
      <Stack direction="row" spacing={1} alignItems="center">
        <TextField
          id="outlined-number"
          label="Every x Minutes"
          value={minutes}
          onChange={onMinuteChange}
          type="number"
          InputLabelProps={{ disabled: cronExpression[4] !== '*', shrink: true, min: 1, max: 59 }}
        />
      </Stack>
    </Stack>
  )
}

export default Minutely
