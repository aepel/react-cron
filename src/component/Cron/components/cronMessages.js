import { defineMessages } from 'react-intl'

export default defineMessages({
  daily: {
    id: 'components.shared.daily',
    defaultMessage: 'Daily',
  },
  onceAYear:{
    id: 'components.shared.onceAYear',
    defaultMessage: 'Once a year',
  },
  yearly: {
    id: 'components.shared.yearly',
    defaultMessage: 'Yearly',
  },
  month:{
    id: 'components.shared.month',
    defaultMessage: 'Month(s)',
  },
  monthly: {
    id: 'components.shared.monthly',
    defaultMessage: 'Monthly',
  },
  quarterly: {
    id: 'components.shared.quarterly',
    defaultMessage: 'Quarterly',
  },
  weekly: {
    id: 'components.shared.weekly',
    defaultMessage: 'Weekly',
  },
  everyWeekDay: {
    id: 'components.shared.everyWeekDay',
    defaultMessage: 'Every week day',
  },
  everyDay: {
    id: 'components.shared.everyDay',
    defaultMessage: 'Every Day(s)',
  },
  hours: {
    id: 'components.shared.hours',
    defaultMessage: 'Hour',
  },
  minutes: {
    id: 'components.shared.minutes',
    defaultMessage: 'Minutes',
  },
  dayOfEveryMonth: {
    id: 'components.shared.dayOfEveryMonth',
    defaultMessage: 'Day of Every Months(s)',
  },

  monday: {
    id: 'components.shared.monday',
    defaultMessage: 'Monday',
  },
  tuesday: {
    id: 'components.shared.tuesday',
    defaultMessage: 'Tuesday',
  },
  wednesday: {
    id: 'components.shared.wednesday',
    defaultMessage: 'Wednesday',
  },
  thursday: {
    id: 'components.shared.thursday',
    defaultMessage: 'Thursday',
  },
  friday: {
    id: 'components.shared.friday',
    defaultMessage: 'Friday',
  },
  saturday: {
    id: 'components.shared.saturday',
    defaultMessage: 'Saturday',
  },
  sunday: {
    id: 'components.shared.sunday',
    defaultMessage: 'Sunday',
  },
  daysBeforeEndMonth: {
    id: 'components.shared.daysBeforeEndMonth',
    defaultMessage: 'Day(s) before the end of the month',
  },
  dayOfTheMonth: {
    id: 'components.shared.dayOfTheMonth',
    defaultMessage: 'Day in the month',
  },
  dayOfEveryQuarter: {
    id: 'components.shared.dayOfEveryQuarter',
    defaultMessage: 'Day of Every Quarter',
  },

  daysBeforeEndQuarter: {
    id: 'components.shared.daysBeforeEndQuarter',
    defaultMessage: 'Day(s) before the end of the Quarter',
  },
  lastWeekdayOfmonth: {
    id: 'components.shared.lastWeekdayOfmonth',
    defaultMessage: 'On the last weekday of every month',
  },
  lastDayOfMonth: {
    id: 'components.shared.lastDayOfMonth',
    defaultMessage: 'Last day of every month',
  },
  lastDayofQuarter: {
    id: 'components.shared.lastDayofQuarter',
    defaultMessage: 'Last day of every Quarter',
  },
  execute: {
    id: 'components.shared.execute',
    defaultMessage: 'Execute',
  },
  everyXMonths: {
    id: 'components.shared.execute',
    defaultMessage: 'Execute',
  },
  1:{
    id: 'components.shared.1',
    defaultMessage: 'January',
  },
  2:{
    id: 'components.shared.2',
    defaultMessage: 'February',
  },
  3:{
    id: 'components.shared.3',
    defaultMessage: 'March',
  },
  4:{
    id: 'components.shared.4',
    defaultMessage: 'April',
  },
  5:{
    id: 'components.shared.5',
    defaultMessage: 'May',
  },
  6:{
    id: 'components.shared.6',
    defaultMessage: 'June',
  },
  7:{
    id: 'components.shared.7',
    defaultMessage: 'July',
  },
  8:{
    id: 'components.shared.8',
    defaultMessage: 'August',
  },
  9:{
    id: 'components.shared.9',
    defaultMessage: 'September',
  },
  10:{
    id: 'components.shared.10',
    defaultMessage: 'October',
  },
  11:{
    id: 'components.shared.11',
    defaultMessage: 'November',
  },
  12:{
    id: 'components.shared.12',
    defaultMessage: 'December',
  },
})
