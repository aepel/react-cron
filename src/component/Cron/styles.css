.Tab_Margin {
  margin-bottom: 10px;
}

.Content {
  margin-left: 12px;
}

.Input_Wrapper {
  margin-bottom: 8px;
}

.Input_Width {
  min-width: 200px;
  width: 200px;
}