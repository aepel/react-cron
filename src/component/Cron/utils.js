import cronstrue from 'cronstrue'
import isArray from 'lodash/isArray'

const getCronHumanValue = (cronExpression, locale) => {
  try {
    if (isArray(cronExpression)) {
      const val = cronstrue.toString(`${cronExpression.join(' ')}`, { verbose: true, locale })

      if (val.search('undefined') === -1) {
        return val
      }
    }

    return '-'
  } catch (ex) {
    console.warn(ex)
    return '-'
  }
}

export { getCronHumanValue }
