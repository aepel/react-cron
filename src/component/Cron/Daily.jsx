import React, { useState } from 'react'
import { FormControlLabel, FormGroup, Radio, Stack, TextField } from '@mui/material'
import { useIntl } from 'react-intl'
import ChooseTime from './components/ChooseTime'
import cronMessages from './components/cronMessages'

function Daily({ cronExpression, onChange }) {
  const { formatMessage } = useIntl()
  const [days, setDays] = useState(1)
  const onDayChange = e => {
    if ((e.target.value > 0 && e.target.value < 32) || e.target.value === '') {
      setDays(parseInt(e.target.value, 10))
      const val = [
        '0',
        cronExpression[1] === '*' ? '0' : cronExpression[1],
        cronExpression[2] === '*' ? '0' : cronExpression[2],
        '*',
        '*',
      ]
      if (e.target.value === '') {
        val[2] = '*'
      } else {
        val[2] = `1/${e.target.value}`
      }

      onChange(val)
    }
  }
  const changeHours = e => {
    const val = [...cronExpression]
    val[1] = `${e.target.value}`
    onChange(val)
  }
  const changeMinutes = e => {
    const val = [...cronExpression]
    val[0] = `${e.target.value}`
    onChange(val)
  }

  return (
    <Stack direction="column" spacing={2} alignItems="flex-start" justifyContent="flex-start">
      <FormGroup row>
        <FormControlLabel
          control={
            <Radio
              color="primary"
              value="1"
              onChange={() => {
                onChange(['0', '0', '1/1', '*', '*'])
              }}
              checked={cronExpression[4] === '*'}
            />
          }
          label={formatMessage(cronMessages.everyDay)}
        />
        <FormControlLabel
          control={
            <Radio
              color="primary"
              value="4"
              onChange={() => {
                onChange([cronExpression[0], cronExpression[1], '*', '*', '1-5'])
              }}
              checked={cronExpression[4] === '1-5'}
            />
          }
          label={formatMessage(cronMessages.everyWeekDay)}
        />
      </FormGroup>
      <Stack direction="row" spacing={1} alignItems="center">
        {cronExpression[4] === '*' ? (
          <TextField
            id="outlined-number"
            label={formatMessage(cronMessages.everyDay)}
            value={days}
            onChange={onDayChange}
            type="number"
            InputLabelProps={{ disabled: cronExpression[4] !== '*', shrink: true }}
            disabled={cronExpression[4] !== '*'}
          />
        ) : null}

        <ChooseTime
          hour={cronExpression[1]}
          minute={cronExpression[0]}
          changeMinutes={changeMinutes}
          changeHours={changeHours}
        />
      </Stack>
    </Stack>
  )
}

export default Daily
