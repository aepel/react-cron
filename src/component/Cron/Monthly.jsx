import React, { useState } from 'react'
import { FormControlLabel, FormGroup, Radio, Stack, TextField } from '@mui/material'
import classNames from 'classnames/bind'
import useI18n from '../../hooks/useI18n/index'
import ChooseTime from './components/ChooseTime'
import styles from './styles.css'

const classes = classNames.bind(styles)

function Monthly({ cronExpression, onChange }) {
  const [every, setEvery] = useState('1')
  const { formatMessage, messages } = useI18n()
  const onDayChange = e => {
    if ((parseInt(e.target.value, 10) > 0 && parseInt(e.target.value, 10) <= 31) || e.target.value === '') {
      const val = [
        cronExpression[0] === '*' ? '0' : cronExpression[0],
        cronExpression[1] === '*' ? '0' : cronExpression[1],
        cronExpression[2],
        '1/1',
        '*',
      ]
      val[2] = `${e.target.value}`
      onChange(val)
    }
  }
  const onEveryXMonth = e => {
    if ((parseInt(e.target.value, 10) > 0 && parseInt(e.target.value, 10) <= 11) || e.target.value === '') {
      const val = [...cronExpression]
      val[3] = `1/${e.target.value}`
      onChange(val)
    }
  }
  const onAtHourChange = e => {
    const val = [...cronExpression]
    val[1] = `${e.target.value}`
    onChange(val)
  }
  const onAtMinuteChange = e => {
    const val = [...cronExpression]
    val[0] = `${e.target.value}`
    onChange(val)
  }

  return (
    <Stack direction="column" spacing={2} alignItems="flex-start" justifyContent="flex-start">
      <Stack direction="row" spacing={1} alignItems="center">
        <div className={classes('Content')}>
          <Stack direction="row" spacing={1} alignItems="center">
            <Stack direction="column" spacing={2} alignItems="flex-start" justifyContent="flex-start">
              <FormGroup>
                <FormControlLabel
                  control={
                    <Radio
                      color="primary"
                      value="1"
                      checked={every === '1'}
                      onChange={() => {
                        setEvery('1')
                        onChange(['0', '0', '1', '*', '*'])
                      }}
                    />
                  }
                  label={formatMessage(messages.dayOfEveryMonth)}
                />
              </FormGroup>
              <FormGroup row>
                <FormControlLabel
                  control={
                    <Radio
                      color="primary"
                      value={cronExpression[2]}
                      onChange={() => {
                        setEvery('2')
                        onChange([
                          cronExpression[0] === '*' ? '0' : cronExpression[0],
                          cronExpression[1] === '*' ? '0' : cronExpression[1],
                          '1',
                          '1/3',
                          '*',
                        ])
                      }}
                      checked={every === '2'}
                    />
                  }
                  label="Every x months"
                />
              </FormGroup>
            </Stack>
            <Stack direction="column" spacing={2} alignItems="flex-end" justifyContent="flex-end" />
          </Stack>
        </div>
      </Stack>
      <Stack direction="row" spacing={1} alignItems="center">
        {every === '1' ? (
          <TextField
            id="outlined-number"
            label={formatMessage(messages.dayOfEveryMonth)}
            value={cronExpression[2]}
            onChange={onDayChange}
            type="number"
            InputLabelProps={{
              shrink: true,
            }}
          />
        ) : (
          <TextField
            style={{ minWidth: '30%' }}
            id="outlined-number"
            label={formatMessage(messages.everyXMonths)}
            value={cronExpression[3].split('/')[1]}
            onChange={onEveryXMonth}
            type="number"
            InputLabelProps={{
              shrink: true,
            }}
            InputProps={{ inputProps: { min: 1, max: 11 } }}
          />
        )}

        {/* {every === '4' ? (
              <TextField
                id="outlined-number"
                label={formatMessage(messages.daysBeforeEndMonth)}
                value={cronExpression[3].split('-')[1]}
                onChange={onLastDayChange}
                type="number"
                className={classes('Input_Width')}
                InputLabelProps={{ readOnly: every !== '4', shrink: true }}
                
                disabled={every !== '4'}
              />
            ) : null} */}
        <ChooseTime
          hour={cronExpression[1]}
          minute={cronExpression[0]}
          changeHours={onAtHourChange}
          changeMinutes={onAtMinuteChange}
        />
      </Stack>
    </Stack>
  )
}

export default Monthly
