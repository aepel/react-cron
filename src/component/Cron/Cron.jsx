import React, { useEffect, useState } from 'react'
import { Card, CardContent, Tab, Tabs } from '@mui/material'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import { useIntl } from 'react-intl'
import cronMessages from './components/cronMessages'
import DisplayCard from './components/DisplayCard'
import Daily from './Daily'
import Minutely from './Minutely'
import Monthly from './Monthly'
import Quarterly from './Quarterly'
import styles from './styles.css'
import { getCronHumanValue } from './utils'
import Weekly from './Weekly'
import Yearly from './Yearly'

const classes = classNames.bind(styles)
const DEFAULT_VALUE = ['0', '0', '*/1', '*', '*']
const options = [
  { name: 'minutely', Component: Minutely, value: ['*/10', '*', '*', '*', '*'] },
  { name: 'daily', Component: Daily, value: DEFAULT_VALUE },
  { name: 'weekly', Component: Weekly, value: ['0', '0', '*', '*', '*'] },
  { name: 'monthly', Component: Monthly, value: ['0', '0', '1', '*', '*'] },
  { name: 'quarterly', Component: Quarterly, value: ['0', '0', '1', '*/3', '*'] },
  { name: 'yearly', Component: Yearly, value: ['0', '0', '1', '1', '*'] },
]

function Cron({ showResultText, showResultCron, onChange, value = null }) {
  const intl = useIntl()
  const { locale, formatMessage } = intl

  const [cronExpression, setCronExpression] = useState(value ?? DEFAULT_VALUE)
  const [selectedTab, setSelectedTab] = useState(1)
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const detectSelectedTab = value => {
    if (!value) return
    if (value[0].indexOf('*/') >= 0) onTabChange(0)
    if (value[4] !== '*') onTabChange(2)
    if (value[3].indexOf('1/') >= 0) onTabChange(3)
    if (value[4] === '1-5') onTabChange(1)
    if (value[2].indexOf('*/') >= 0) onTabChange(1)
    if (value[3].indexOf('*/3') >= 0) onTabChange(4)
    if (value[4].indexOf('*/3') >= 0) onTabChange(4)
    if (value[3].indexOf('/') === -1 && value[3] !== '*') onTabChange(5)
  }
  useEffect(() => {
    if (value && value === cronExpression) {
      detectSelectedTab(value)
    }
    // eslint-disable-next-line no-console
  }, [detectSelectedTab, value, cronExpression])
  const onTabChange = idx => {
    setSelectedTab(idx)
    if (value && value === cronExpression) {
      onChange(value, getCronHumanValue(value, locale), options[idx].name)
    } else {
      setCronExpression(options[idx].value)
      onChange(options[idx].value, getCronHumanValue(options[idx].value, locale), options[idx].name)
    }
  }

  const onValueChange = val => {
    if (val && val.length) {
      setCronExpression(val)
    } else {
      setCronExpression(['0', '0', '1/1', '*', '*'])
      val = ['0', '0', '1/1', '*', '*']
    }

    onChange(val, getCronHumanValue(val, locale), options[selectedTab].name)
  }
  const Selector = options[selectedTab].Component

  return (
    <Card style={{ maxWidht: '100%', minWidth: '25%' }}>
      <CardContent>
        <Tabs value={selectedTab} className={classes('Tab_Margin')}>
          {options.map((option, idx) => (
            <Tab
              fullWidth
              key={idx}
              label={formatMessage(cronMessages[option.name])}
              onClick={() => onTabChange(idx)}
            />
          ))}
        </Tabs>

        <Selector key={options[selectedTab].name} cronExpression={cronExpression} onChange={onValueChange} />
        <DisplayCard
          textResult={showResultText && getCronHumanValue(cronExpression, locale)}
          cronResult={showResultCron && cronExpression.join(' ')}
        />
      </CardContent>
    </Card>
  )
}

Cron.propTypes = {
  showResultText: PropTypes.bool,
  showResultCron: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.arrayOf(PropTypes.string),
}

Cron.defaultProps = {
  showResultText: true,
  showResultCron: false,
  value: null,
}
export default Cron
