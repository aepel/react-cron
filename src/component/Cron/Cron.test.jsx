import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { IntlProvider } from 'react-intl'
import cronMessages from './components/cronMessages'
import Cron from './Cron'

const props = {
  showResultText: false,
  showResultCron: true,
  onChange: () => true,
}

describe('Cron Component', () => {
  describe('snapshot test', () => {
    it('should shallow render export list correctly', () => {
      const { asFragment } = render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
  describe('Weekly', () => {
    it('Should return default cron expression for daily', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Weekly'))
      expect(screen.getByText('(0 0 1/1 * *)')).toBeInTheDocument()
    })

    it('Should return default cron expression for Every Tuesday and friday', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Weekly'))
      await userEvent.click(screen.getByText('Tuesday'))
      await userEvent.click(screen.getByText('Monday'))
      expect(screen.getByText('(0 0 1/1 * MON,TUE)')).toBeInTheDocument()
    })
  })

  describe('Quarterly', () => {
    it('Should return default cron expression for Quarterly', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Quarterly'))
      expect(screen.getByText('(0 0 1 */3 *)')).toBeInTheDocument()
    })
  })
  describe('Monthly', () => {
    it('Should return default cron expression for Day of Every Months(s)', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Monthly'))

      expect(screen.getByText('(0 0 1 * *)')).toBeInTheDocument()
    })
  })
  describe('Daily', () => {
    it('Should return default cron expression for Every Day(s)', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Daily'))

      expect(screen.getByText('(0 0 1/1 * *)')).toBeInTheDocument()
    })
    it('Should return default cron expression for Every week day', async () => {
      render(
        <IntlProvider messages={cronMessages} locale="en" defaultLocale="en">
          <Cron {...props} />
        </IntlProvider>
      )
      await userEvent.click(screen.getByText('Daily'))
      await userEvent.click(screen.getByText('Every week day'))

      expect(screen.getByText('(0 0 * * 1-5)')).toBeInTheDocument()
    })
  })
})
