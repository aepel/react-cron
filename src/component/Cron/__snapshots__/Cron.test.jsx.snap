// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cron Component snapshot test should shallow render export list correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="min-width: 25%;"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTabs-root Tab_Margin css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Daily
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Weekly
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Monthly
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Quarterly
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Yearly
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="css-hyolpn-MuiStack-root"
      >
        <div
          class="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value="1"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Every Day(s)
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
            >
              <input
                class="PrivateSwitchBase-input css-1m9pwf3"
                type="radio"
                value="4"
              />
              <span
                class="css-hyxlzm"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                  data-testid="RadioButtonCheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Every week day
            </span>
          </label>
        </div>
        <div
          class="css-1q36rmd-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="outlined-number"
              id="outlined-number-label"
            >
              Every Day(s)
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-number"
                type="number"
                value="1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Every Day(s)
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Hour
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Input_Width css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="select-wrapper-chooseTime-hour"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                00
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                data-testid="select-chooseTime-hour"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Hour
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="simple-select-minutes"
            >
              Minutes
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Input_Width css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                00
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="0"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Minutes
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        (0 0 1/1 * *)
      </p>
    </div>
  </div>
</DocumentFragment>
`;
